platform/linux/linux.cmake

```diff
@@ -4,11 +4,21 @@ find_package(ICU OPTIONAL_COMPONENTS uc)
 find_package(JPEG REQUIRED)
 find_package(PNG REQUIRED)
 find_package(PkgConfig REQUIRED)
-find_package(X11 REQUIRED)
+if (MBGL_WITH_X11_SUPPORT)
+    find_package(X11 REQUIRED)
+endif ()
 find_package(Threads REQUIRED)
 
 pkg_search_module(LIBUV libuv REQUIRED)
 
+if (MBGL_WITH_WAYLAND_SUPPORT)
+    add_compile_definitions(
```

Comment:
We try to avoid global definitions. Are they really needed or can they be added to the relevant target (maybe even `mbgl-compile-options` would do)?

Reply:
I was avoiding having to define them twice.  Switched to target_compile_definitions for each

Reply:
If you define them as `PUBLIC` they will get propagated with dependencies.

Reply:
yes I'm very familiar with CMake

Reply:
Sorry, I just wanted to be helpful 😊